src/app/components/Modal/Modal.ts

```diff
@@ -0,0 +1,247 @@
+          this.errFields.get('cvc')?.node.classList.add('card-error_active');
+        } else if (i === 5) {
+          this.errFields.get('cvc')?.node.classList.remove('card-error_active');
+        }
+      });
+
+      if (!validationArr.length && validation) {
+        (this.formInfo.node as HTMLFormElement).reset();
+        const items = CartController.getCartItems();
+        items?.forEach((elem) => {
+          CartController.remove(elem.id);
+        });
+        overlay.node.classList.add('overlay_active');
+
+        setTimeout(() => {
+          this.node.classList.remove('modal-active');
+          window.location.hash = 'products';
+          this.removeModal(app);
+        }, 3000);
+      }
+    };
+
+    this.node.onclick = (e) => {
+      if (e.target === this.node) {
+        this.node.classList.remove('modal-active');
+        this.removeModal(app);
+        (this.formInfo.node as HTMLFormElement).reset();
+      }
+    };
+  }
+
+  removeModal = (app: App | undefined) => {
+    if (app) {
+      app.main.node.classList.remove('modal-open');
+      document.documentElement.classList.remove('modal-open');
+    }
+  };
+
+  validateUserField = (fieldName: string, fieldVal: string) => {
+    let isValid = true;
+    if (fieldName === 'name' || fieldName === 'address') {
+      isValid = this.validateNameOrAddress(fieldName, fieldVal);
+    }
+    if (fieldName === 'phone' || fieldName === 'email') {
+      isValid = this.validatePhoneOrEmail(fieldName, fieldVal);
+    }
+    return isValid;
+  };
+
+  checkWordsLengthValidity = (str: string, minWordLength: number) => {
+    const words = str.trim().split(' ');
+    const regExp = minWordLength === 5 ? /[А-Яа-яA-Za-z]{5,}$/ : /[А-Яа-яA-Za-z]{3,}/;
+    const validWords = words.filter((word) => regExp.test(word));
+    return words.length === validWords.length;
+  };
+
+  checkWordsNumValidity = (str: string, minWordsNum: number) => {
+    return str.trim().split(' ').length >= minWordsNum;
+  };
+
+  validateCardNumber = () => {
+    let digitsNum = 0;
+    this.cardData.forEach((input, i) => {
+      if (i <= 3) {
+        const node = input.node as HTMLInputElement;
+        digitsNum += node.value.length;
+      }
+    });
+    if (digitsNum !== 16) {
+      this.errFields.get('card number')?.node.classList.add('card-error_active');
+    } else {
+      this.errFields.get('card number')?.node.classList.remove('card-error_active');
+    }
+    return digitsNum === 16;
+  };
+
+  validatePhoneOrEmail = (fieldName: string, fieldVal: string) => {
+    const pattern = fieldName === 'phone' ? /^\+\d{9,}$/ : /^[^ ]+@[^ ]+\.[a-z]{2,3}$/;
+    return pattern.test(fieldVal);
+  };
+
+  validateNameOrAddress = (fieldName: string, fieldVal: string) => {
+    const [numLimit, lengthLimit] = fieldName === 'name' ? [2, 3] : [3, 5];
+    if (
+      !this.checkWordsNumValidity(fieldVal, numLimit) ||
+      !this.checkWordsLengthValidity(fieldVal, lengthLimit)
+    ) {
+      return false;
```

Comment:
return !(!this.checkWordsNumValidity(fieldVal, numLimit) ||
        !this.checkWordsLengthValidity(fieldVal, lengthLimit));